iP/src/duke/TaskList.java

```diff
@@ -47,4 +47,14 @@ public Task getTask(int index) {
         Task task = taskList.get(index);
         return task;
     }
+
+    public ArrayList<Task> find(String keyword) {
```

Comment:
I like how you return your output as an ArrayList<Task> to be processed later instead of immediately returning a processed string since this could improve adaptability and extensibility of your program! 👍

Reply:
I agree